google-cloud-spanner/test/google/cloud/spanner/convert/grpc_value_to_object_test.rb

```diff
@@ -134,4 +134,20 @@
     raw = Google::Cloud::Spanner::Convert.grpc_value_to_object value, type
     _(raw).must_equal BigDecimal(number)
   end
+
+  it "converts a JSON value" do
+    venue_detail = { "name" => "ABC", "open" => true, "rating" => 10 }
+
+    value = Google::Protobuf::Value.new(string_value: venue_detail.to_json)
+    type = Google::Spanner::V1::Type.new(code: :JSON)
+    raw = Google::Cloud::Spanner::Convert.grpc_value_to_object value, type
+    _(raw).must_equal venue_detail
+  end
+
+  it "converts empty JSON value to nil" do
```

Comment:
I don't think we will get an empty string, which is not a valid JSON value. If it does, we should get an error.